test/functional/p2p_local_tx_relay.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""
+Test how locally submitted transactions are sent to the network when sensitive relay is used.
+
+The topology in the test is:
+
+Bitcoin Core (node0)
+               ^   v       The default full-outbound + block-only connections
+               |   |       (MAX_OUTBOUND_FULL_RELAY_CONNECTIONS + MAX_BLOCK_RELAY_ONLY_CONNECTIONS)
+               |   |
+               |   *-----> SOCKS5 Proxy ---> P2PInterface (self.socks5_server.conf.destinations[0]["node"])
+               |   |
+               |   *-----> SOCKS5 Proxy ---> P2PInterface (self.socks5_server.conf.destinations[1]["node"])
+               |   ...
+               |   |       The sensitive relay TX recipients (SENSITIVE_RELAY_NUM_BROADCAST_PER_TX):
+               |   |
+               |   *-----> SOCKS5 Proxy ---> P2PInterface (self.socks5_server.conf.destinations[i]["node"])
+               |   |
+               |   *-----> SOCKS5 Proxy ---> P2PInterface (self.socks5_server.conf.destinations[i + 1]["node"])
+               |   ...
+               |
+               *---------< observer_inbound
+"""
+
+from test_framework.p2p import P2PDataStore, P2PInterface, P2P_SERVICES
+from test_framework.messages import msg_getdata, msg_inv, msg_mempool, CInv, MSG_WTX
+from test_framework.socks5 import Socks5Configuration, Socks5Server
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, MAX_NODES, p2p_port
+from test_framework.wallet import MiniWallet
+
+MAX_OUTBOUND_FULL_RELAY_CONNECTIONS = 8
+MAX_BLOCK_RELAY_ONLY_CONNECTIONS = 2
+SENSITIVE_RELAY_NUM_BROADCAST_PER_TX = 5
+
+g_p2p_index = None
+def new_p2p_index():
+    global g_p2p_index
```

Comment:
I think there is no need of creating a global variable like `g_p2p_index`. We could do something like:
```diff
diff --git a/test/functional/p2p_local_tx_relay.py b/test/functional/p2p_local_tx_relay.py
index 0b115352d..9f4c6aaa5 100755
--- a/test/functional/p2p_local_tx_relay.py
+++ b/test/functional/p2p_local_tx_relay.py
@@ -36,12 +36,6 @@ MAX_OUTBOUND_FULL_RELAY_CONNECTIONS = 8
 MAX_BLOCK_RELAY_ONLY_CONNECTIONS = 2
 SENSITIVE_RELAY_NUM_BROADCAST_PER_TX = 5
 
-g_p2p_index = None
-def new_p2p_index():
-    global g_p2p_index
-    g_p2p_index += 1
-    return g_p2p_index
-
 # Fill addrman with these addresses. Must have enough Tor addresses, so that even
 # if all 10 default connections are opened to a Tor address (!?) there must be more
 # for sensitive relays.
@@ -135,13 +129,16 @@ class P2PLocalTxRelay(BitcoinTestFramework):
     def set_test_params(self):
         self.disable_autoconnect = False
         self.num_nodes = 1
-        global g_p2p_index
-        g_p2p_index = self.num_nodes - 1
+        self.g_p2p_index = self.num_nodes - 1
+
+    def new_p2p_index(self):
+        self.g_p2p_index += 1
+        return self.g_p2p_index
 
     def setup_nodes(self):
         # Start a SOCKS5 proxy server.
         socks5_server_config = Socks5Configuration()
-        socks5_server_config.addr = ("127.0.0.1", p2p_port(new_p2p_index()))
+        socks5_server_config.addr = ("127.0.0.1", p2p_port(self.new_p2p_index()))
         socks5_server_config.unauth = True
         socks5_server_config.auth = True
```

Reply:
I removed `new_p2p_index()` because it was used in just one place. Earlier it was used also for all listeners behind the SOCKS5 proxy, but I had to stop using `p2p_port()` for that because it exceeded `MAX_NODES` (12).